src/slice/mod.rs

```diff
@@ -557,6 +611,67 @@ impl<'data, T: Sync + 'data> IndexedParallelIterator for Chunks<'data, T> {
     }
 }
 
+/// Parallel iterator over immutable non-overlapping chunks of a slice
+#[derive(Debug)]
+pub struct ChunksExact<'data, T: 'data + Sync> {
+    chunk_size: usize,
+    slice: &'data [T],
+    rem: &'data [T],
+}
+
+impl<'data, T: Sync> ChunksExact<'data, T> {
+    /// Return the remainder of the original slice that is not going to be
+    /// returned by the iterator. The returned slice has at most `chunk_size-1`
+    /// elements.
+    pub fn remainder(&self) -> &'data [T] {
+        self.rem
+    }
+}
+
+impl<'data, T: Sync> Clone for ChunksExact<'data, T> {
```

Comment:
Isn't this exactly what `#[derive(Clone)]` would generate?

Reply:
Yes. I have no idea why I wrote this, I must have been half asleep at the time.

Reply:
I think `#[derive(Clone)]` would also force `T: Clone`, which is not needed to clone the slice.

Reply:
I think you're both right, sorry :)